irohad/wsv_checker/wsv_checker.cpp

```diff
@@ -0,0 +1,1119 @@
+      fmt::print(name " have different " STR(x) ": '{}' and '{}'\n", (x), (y));                                        \
+   }
+#define CHECK_EQUALS_RETURN(name, x, y)                                                                                \
+   if ((x) != (y)) {                                                                                                   \
+      fmt::print(name " have different " STR(x) ": '{}' and '{}'\n", (x), (y));                                        \
+      return false;                                                                                                    \
+   }
+#define CHECK_EQUALS_2(type, name, x, y)                                                                               \
+   if ((x) != (y)) {                                                                                                   \
+      fmt::print(type " {} have different " STR(x) ": '{}' and '{}'\n", (name), (x), (y));                             \
+   }
+#define CHECK_EQUALS_STR(type, name, x, y)                                                                             \
+   if ((x) != (y)) {                                                                                                   \
+      auto sx = short_string(x);                                                                                       \
+      auto sy = short_string(y);                                                                                       \
+      fmt::print(type " {} have different " STR(x) ": '{}{}' and '{}{}'\n", (name), sx.str, sx.dots, sy.str, sy.dots); \
+   }
+#define CHECK_EQUALS_JSON(type, name, x, y)                                                                            \
+   if ((x) != (y)) {                                                                                                   \
+      auto sx = shortstring((x).dump());                                                                               \
+      auto sy = shortstring((y).dump());                                                                               \
+      fmt::print(type " {} have different " STR(x) ": '{}' and '{}'\n", (name), sx, sy);                               \
+   }
+#define COUNT_INEQUALITIES(type, name, x, y)                                                                           \
+   if ((x) != (y)) {                                                                                                   \
+      ++inequalities_counter;                                                                                          \
+      fmt::print(type " {} have different " STR(x) ": '{}' and '{}'\n", (name), (x), (y));                             \
+   }
+
+auto
+short_string(std::string const& x)
+{
+   // clang-format off
+   struct{string_view str, dots;} out;
+   // clang-format on
+   out.str = string_view(x).substr(0, 80);
+   out.dots = out.str.size() < x.size() ? "..." : "";
+   return out;
+}
+// clang-format off
```

Comment:
Remove plz.

Reply:
I use it to write more compact code. Sometimes formatted code looks weird.

Reply:
done